gnucash/import-export/aqb/gnc-flicker-gui.c

```diff
@@ -0,0 +1,448 @@
+
+/* structured data for the flicker variables */
+struct Flickerdraw {
+    const char *challenge;
+    guint challenge_length;
+    guint margin;           /* Distance between bars */
+    guint barwidth;         /* Bar width */
+    guint barheight;        /* Bar height */
+    guint x_barpos;         /* x-value for the position of the bar */
+    guint y_barpos;         /* y-value for the position of the bar */
+    guint x_drawpos;        /* x-value of the first painting position */
+    guint y_drawpos;        /* y-value of the first painting position */
+    guint height;           /* Height of the drawing area */
+    guint width;            /* Width of the drawing area */
+    guint delay;            /* Waiting time between frames in milliseconds */
+    guint halfbyteid;
+    guint clock;
+    guint interval;
+    gboolean change_interval;
+};
+static struct Flickerdraw flickerdraw;
+
+static GncFlickerGui *flickergui = NULL;
+static _Bool bitarray[255][5];
+
+/* this function will return number corresponding 0,1,2..,9,A,B,C,D,E,F */
+static uint
+get_num (const char ch)
+{
+    int num =0;
+    if (ch >= '0' && ch <= '9')
+        num = ch - 0x30;
+    else
+    {
+        switch (ch)
+        {
+            case 'A': case 'a': num = 10; break;
+            case 'B': case 'b': num = 11; break;
+            case 'C': case 'c': num = 12; break;
+            case 'D': case 'd': num = 13; break;
+            case 'E': case 'e': num = 14; break;
+            case 'F': case 'f': num = 15; break;
+            default: num = 0;
+        }
+    }
+    /* The bank challenge has been verified by Aqbanking,
+     * so I assume that no error can occur. */
+    return num;
+}
+
+/* convert the bank challenge into the 5 bits for the flicker data */
+static char
+*flicker_data (const char *challenge)
+{
+    /* bitfield is a clock bit and a 4-bit code with the bits reversed
+       (bit 1 is the least significant and bit 4 the most
+       so 0x1 is 1000 and 0x8 is 0001) */
+     static const _Bool bits[16][5] =
+     {
+           {0, 0, 0, 0, 0}, {0, 1, 0, 0, 0}, {0, 0, 1, 0, 0}, {0, 1, 1, 0, 0},
+           {0, 0, 0, 1, 0}, {0, 1, 0, 1, 0}, {0, 0, 1, 1, 0}, {0, 1, 1, 1, 0},
+           {0, 0, 0, 0, 1}, {0, 1, 0, 0, 1}, {0, 0, 1, 0, 1}, {0, 1, 1, 0, 1},
+           {0, 0, 0, 1, 1}, {0, 1, 0, 1, 1}, {0, 0, 1, 1, 1}, {0, 1, 1, 1, 1}
+     };
+
+     /* prepend synchronization identifier */
+     const char pre[] = {'0', 'F', 'F', 'F'};
+     size_t len = sizeof (pre) + strlen (challenge) + 1;
+     char* code = (char*)malloc (len);
+     memcpy (code, pre, sizeof (pre));
+     memcpy (code + sizeof (pre), challenge, strlen (challenge));
+
+    /* Swap the position of the bits in pairs throughout the bank challenge
+       (low-order nibble first). */
+     for (uint i = 0; i < len - 1; i += 2)
+     {
+         u_int val1 = get_num (code[i]);
+         u_int val2 = get_num (code[i+1]);
+
+         if (val1 < 0x10 && val2 < 0x10)
```

Comment:
You don't need the `if`, the condition is always true.

Reply:
Done.